### PR TITLE
[HWKMETRICS-317] Add @GET on root for each metric type to match the e…

### PR DESCRIPTION
**Pull request by stefannegrea**
…nd-point for adding metrics via @POST.

### CODE REVIEW

**Comment by stefannegrea** [comment]
retest this please

File: api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/AvailabilityHandler.java [hunk 1]
```diff
@@ -115,6 +116,39 @@ public void createAvailabilityMetric(
...
+                    notes = "Does not include any metric values. ",
+                    response = MetricDefinition.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved at least one metric definition."),
+            @ApiResponse(code = 204, message = "No metrics found."),
+            @ApiResponse(code = 400, message = "Invalid type parameter type.", response = ApiError.class),
+            @ApiResponse(code = 500, message = "Failed to retrieve metrics due to unexpected error.",
+                    response = ApiError.class)
+    })
+    public void findGuageMetrics(
```
Comment (burmanm) [inline api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/AvailabilityHandler.java]: GuageMetrics? AvailabilityMetrics perhaps?

File: api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/CounterHandler.java [hunk 2]
```diff
@@ -122,6 +123,39 @@ public void createCounter(
...
+                    notes = "Does not include any metric values. ",
+                    response = MetricDefinition.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved at least one metric definition."),
+            @ApiResponse(code = 204, message = "No metrics found."),
+            @ApiResponse(code = 400, message = "Invalid type parameter type.", response = ApiError.class),
+            @ApiResponse(code = 500, message = "Failed to retrieve metrics due to unexpected error.",
+                    response = ApiError.class)
+    })
+    public void findGuageMetrics(
```
Comment (burmanm) [inline api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/CounterHandler.java]: findCounterMetrics?

File: api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/GaugeHandler.java [hunk 3]
```diff
@@ -117,6 +119,39 @@ public void createGaugeMetric(
...
+                    notes = "Does not include any metric values. ",
+                    response = MetricDefinition.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved at least one metric definition."),
+            @ApiResponse(code = 204, message = "No metrics found."),
+            @ApiResponse(code = 400, message = "Invalid type parameter type.", response = ApiError.class),
+            @ApiResponse(code = 500, message = "Failed to retrieve metrics due to unexpected error.",
+                    response = ApiError.class)
+    })
+    public void findGuageMetrics(
```
Comment (burmanm) [inline api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/GaugeHandler.java]: findGaugeMetrics?

**Comment by burmanm** [comment]
LGTM, squash/fixup the second commit and merge

**Comment by stefannegrea** [comment]
@burmanm, I just rebased on master. Ready for merge!

**Comment by burmanm** [comment]
@stefannegrea squash ;) The fixups are not nice in the long run in commit list (or if one needs to check blame/history to see who knows about the change)

**Comment by stefannegrea** [comment]
@burmanm, not a big fan of squashing but did it :)

**Comment by stefannegrea** [comment]
@burmanm, the results of the performance job can be ignored for now.

**Pull request merged**